ossindex/ossindex.go

```diff
@@ -181,7 +184,13 @@ func (o *Server) setupRequest(jsonStr []byte) (req *http.Request, err error) {
 	o.logLady.WithField("user_agent", ua).Debug("Obtained User Agent for request to OSS Index")
 
 	req.Header.Set("Content-Type", "application/json")
-	if o.Options.Username != "" && o.Options.Token != "" {
+	if o.Options.Username != "" || o.Options.Token != "" {
+		switch {
+		case o.Options.Token == "":
+			return nil, fmt.Errorf(msgMissingOssiToken)
+		case o.Options.Username == "":
+			return nil, fmt.Errorf(msgMissingOssiUsername)
+		}
```

Comment:
This feels a little weird. To check that neither are not empty and then check if they are empty. Feels like we should check the Username and Token long before we do any of the other setup and return at that point. And then this `!= ""` if check just goes away

Reply:
I share the odd feeling about "double checking" the same two field values, but not sure I see a better way to do it. The "default" case is nothing is set (using non-registered ossi request), so we only care about auth if either field is set, and then we need both set.
I smell the smell, but can't find the air freshener. 
Happy to clean it up later if I my light bulb lights.

Regarding checking this earlier, not sure I see a better place, as this is the first place we look into the stuff that is only needed to for the request. Sorry if I'm missing your point.

Reply:
Yah yah that's right ... You can ignore me forgot it was empty by default and optional. Carry on